src/libsyntax/ext/mbe/macro_rules.rs

```diff
@@ -1184,3 +1177,15 @@ impl TokenTree {
         parse(cx.parse_sess(), tts, mtch, Some(directory), true)
     }
 }
+
+/// Generates an appropriate parsing failure message. For EOF, this is "unexpected end...". For
+/// other tokens, this is "unexpected token...".
+fn parse_failure_msg(tok: &Token) -> String {
```

Comment:
why not
```rust
fn parse_failure_msg(tok: &Token) -> String {
    if let token::Eof = tok.kind {
        return "unexpected end of macro invocation".to_string();
    }
    format!("no rules expected the token `{}`", pprust::token_to_string(tok))
}
```

Reply:
I just haven't looked into the function body at all :-) I think that the best way to format it would probably be

```rust
fn parse_failure_msg(tok: &Token) -> String {
    match tok.kind {
        token::Eof => "unexpected end of macro invocation".to_string(),
        _ => format!("no rules expected the token `{}`", pprust::token_to_string(tok)),
    }
}
```

Not a fan of `if let` with an else case